ecs-cli/modules/utils/regcreds/creds_input_reader.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2015-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package readers
+
+import (
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+)
+
+// ECSRegCredsInput contains registry cred entries for creation and/or use in a task execution role
+type ECSRegCredsInput struct {
+	Version             string
+	RegistryCredentials RegistryCreds `yaml:"registry_credentials"`
+}
+
+// RegistryCreds is a map of registry names to RegCredEntry structs
+type RegistryCreds map[string]RegistryCredEntry
+
+// RegistryCredEntry contains info needed to create an AWS Secrets Manager secret and match it to an ECS container(s)
+type RegistryCredEntry struct {
+	SecretManagerARN string   `yaml:"secret_manager_arn"`
+	Username         string   `yaml:"username"`
+	Password         string   `yaml:"password"`
+	KmdKeyID         string   `yaml:"kms_key_id"`
+	ContainerNames   []string `yaml:"container_names"`
+}
+
+// ReadCredsInput parses 'registry-creds up' input into an ECSRegCredsInput struct
+func ReadCredsInput(filename string) (*ECSRegCredsInput, error) {
+	if filename == "" {
+		return nil, errors.New("file is required")
```

Comment:
I think it would make more sense to validate if the filename is empty or not outside of this function. This function should just read the input file

Reply:
after some testing, it looks like you can't pass in an empty value to the `--file` flag w/o urfav/cli yelling at you, so I agree this check looks unnecessary.

Reply:
See above comment about changing this to a command arg rather than a flag. It's probably yelling at you for not passing in a flag value because you specify this to be a string flag, and `--file` with nothing after would be interpreted as a boolean flag which isn't quite accurate, even though we are getting an "error" of sorts for a missing parameter.